core/src/main/java/com/microsoft/applicationinsights/internal/perfcounter/JniPCConnector.java

```diff
@@ -212,21 +211,11 @@ private static void extractToLocalFolder(File dllOnDisk, String libraryToLoad) t
     }
 
     private static File buildDllLocalPath() {
-        Properties properties = PropertyHelper.getSdkVersionProperties();
-        if (properties == null) {
-            throw new RuntimeException("Failed to find SDK Version Properties file.");
-        }
-
-        String version = properties.getProperty("version");
-        if (version == null) {
-            throw new RuntimeException("Failed to find SDK version.");
-        }
-
         File dllPath = LocalFileSystemUtils.getTempDir();
 
         dllPath = new File(dllPath.toString(), AI_BASE_FOLDER);
         dllPath = new File(dllPath.toString(), AI_NATIVE_FOLDER);
-        dllPath = new File(dllPath.toString(), version);
+        dllPath = new File(dllPath.toString(), PropertyHelper.getSdkVersionNumber()); // FIXME throw if version is unknown?
```

Comment:
I would prefer marking as "unknown" rather than throwing. SDK should still send valuable telemetry

Reply:
The change here is that it will be marked as `"java:unknown"`, so we can tell that it came from Java SDK. Before it was `"unknown"`, without the qualifier.

Reply:
ah got it makes sense. Thanks for clarification @littleaj

Reply:
actually, here it will just be "unknown" because `:` is not a valid file name character...